Unit.cpp

```diff
@@ -15,14 +16,35 @@ std::string Unit::getName() const
 	return name;
 }
 
-void Unit::takeDamage(const Unit& enemy)
+void Unit::takeDamage(Unit& enemy)
 {
+	int dmg_taken = hp;
 	int damage = enemy.getDmg();
 	hp -= damage;
 	if (hp < 0)
 	{
 		hp = 0;
 	}
+	dmg_taken = dmg_taken - hp;
+	enemy.boost_xp(dmg_taken);
+	
+}
+void Unit::boost_xp(const int& dmg)
+{
+	current_xp += dmg;
+	float level_xp = current_xp/100.0;
+	if(level_xp >= level)

```

Comment:
Ez mindig igaz lesz, nem?

Reply:
Itt szerintem rosszul neveztem el a változót, ezért ezt javítottam is `current_xp` -> `xp`
A `level` az már 1 értékkel indul(én úgy gonodltam hogy az első szintről indulunk), ezért nem mindig igaz.

Reply:
Viszont ha a `level` 1-ről indul, akkor itt kéne egy `+1` még: `float level_xp = xp/100.0 + 1;` És akkor tényleg azt számolná ki, hogy hányas szintűnek kéne lennie az xp alapján. Mert most a 100 xp-re 1 jön ki, pedig akkor már 2-es szintre kéne lépnie. Vagy elnéztem valamit?

Reply:
`./a.out test/units/unit2.json test/units/unit3.json` -nek ki printeltem: a szint ugrásait:

`Level_xp: 1.5 Level: 1`\
`Sully is stepped to: 2 level with:150 xp`\
`Level_xp: 2.33 Level: 2`\
`Sully is stepped to: 3 level with:233 xp`\
`Sully wins. Remaining HP: 288`\

Nekem jónak tűnik a működése, de lehet én néztem be valamit(megint)  :)
`

Reply:
Áh, most látom, hogy lentebb a for ciklusban `<=` van, nem `<`, akkor azt néztem el. Én jobb szeretem, ha 0-tól <n-ig megy, és akkor n-szer fut le. Így n+1-szer fut le, tehát ott van az a +1, amit hiányoltam.